Samples/7.ServerList/Program.cs

```diff
@@ -76,44 +77,8 @@ static void Main( string[] args )
                 }
             }
 
-            if ( File.Exists( "servers.bin" ) )
-            {
-                // last time we connected to Steam, we got a list of servers. that list is persisted below.
-                // load that list of servers into the server list.
-                // this is a very simplistic serialization, you're free to serialize the server list however
-                // you like (json, xml, whatever).
-
-                using ( var fs = File.OpenRead( "servers.bin" ) )
-                using ( var reader = new BinaryReader( fs ) )
-                {
-                    while ( fs.Position < fs.Length )
-                    {
-                        var numAddressBytes = reader.ReadInt32();
-                        var addressBytes = reader.ReadBytes( numAddressBytes );
-                        var port = reader.ReadInt32();
-
-                        var ipaddress = new IPAddress( addressBytes );
-                        var endPoint = new IPEndPoint( ipaddress, port );
-
-                        CMClient.Servers.TryAdd( endPoint );
-                    }
-                }
-
-                Console.WriteLine($"Loaded {CMClient.Servers.GetAllEndPoints().Length} servers from server list cache.");
-            }
-            else
-            {
-                // since we don't have a list of servers saved, load the latest list of Steam servers
-                // from the Steam Directory.
-                var loadServersTask = SteamDirectory.Initialize( cellid );
-                loadServersTask.Wait();
-
-                if ( loadServersTask.IsFaulted )
-                {
-                    Console.WriteLine( "Error loading server list from directory: {0}", loadServersTask.Exception.Message );
-                    return;
-                }
-            }
+            SteamClient.Servers.CellID = cellid;
+            SteamClient.Servers.ServerListProvider = new FileStorageServerListProvider("servers.bin");
```

Comment:
Suggest using a different name here - for anyone who has run Sample7 before, the existing data will be incompatible.